### DISCUSSION TITLE
guide-R/manipulation/combiner-variables

### DISCUSSION THREAD

**Discussion by giscus** [larmarange/guide-R#34] [Forum]
# guide-R/manipulation/combiner-variables

https://larmarange.github.io/guide-R/manipulation/combiner-variables.html

**Comment by psychiatrie-nicolas-cazayous** [comment]
Bonjour,
Vous présentez `case_when()` et juste après `labelled::recode_if()`.
Je ne suis pas sûr de voir le bénéfice par rapport à `case_when()`.
```r
df |>
  mutate(
    pref = case_when(
      autre_details == "bleu ciel" ~ "bleu",
      TRUE ~ pref))
```
Merci pour votre package `labelled` et le `guide-R`

**Reply by larmarange** [reply]
Dans le cas présent, avec `case_when()`, la variable *pref* devient une variable de type *character*. En effet, c'est le type de la première modalité du case_when() qui prend le dessus.

Avec `recode_if()`, la variable originale n'est pas transformée et le résultat reste donc un facteur. Tout est une question de préservation des attributs de la variable.

En résumé, `case_when()` ou `if_else()` pour créer une variable, `recode_if()` pour mettre à jour certaines modalités de la variable (en s'assurant de la préservation des attributs originaux).

Cordialement
